torch/csrc/jit/graph_executor.cpp

```diff
@@ -441,11 +442,23 @@ struct GraphExecutorImpl {
     {
       std::lock_guard<std::mutex> lock(compile_mutex);
       auto it = plan_cache.find(spec);
-      if (it != plan_cache.end())
-        return it->second;
-      auto plan = compileSpec(spec);
-      auto r = plan_cache.emplace(std::move(spec), std::move(plan));
-      return r.first->second;
+      if (it == plan_cache.end())
+      {
+        profiling_record_.profiling_count_ = 3;
+        auto plan = compileSpec(spec);
+        it = plan_cache.emplace(spec, std::move(plan)).first;
+      }
+
+      auto& plan = it->second;
+      if (profiling_record_.profiling_count_ == 0) {
+        //re-compile with profiling info
+        auto new_plan = compileSpec(spec);
+        plan_cache[spec] = std::move(plan);
+        profiling_record_.profiling_count_ = -1;
+        return plan_cache[spec];
```

Comment:
Nit: can return `plan_cache[spec]` in both `if` and `else`, or even better: remove the `else`, return after the `if`.